integration_tests/test_str_01.py

```diff
@@ -45,6 +45,21 @@ def test_str_repeat():
     assert 3*a*3 == "XyzXyzXyzXyzXyzXyzXyzXyzXyz"
     assert a*-1 == ""
 
+def test_str_join():
+    a: str
+    a = ","
+    p:list[str] = ["a","b"]
+    res:str = a.join(p)
+    assert res == "a,b"
+
+def test_str_join2():
+    a: str
+    a = "**"
+    p:list[str] = ["a","b"]
+    res:str = a.join(p)
+    assert res == "a**b"
```

Comment:
We can test more cases like empty strings, more items in the list (more than 3), and so on...
This can be done in the subsequent PRs.